src/main/java/appeng/core/sync/packets/PacketCompassRequest.java

```diff
@@ -20,65 +20,98 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
+import net.minecraft.client.Minecraft;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+
 import appeng.api.util.DimensionalCoord;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.core.sync.network.NetworkHandler;
 import appeng.core.worlddata.WorldData;
 import appeng.services.compass.ICompassCallback;
 
 
-public class PacketCompassRequest extends AppEngPacket implements ICompassCallback
+public class PacketCompassRequest implements AppEngPacket, AppEngPacketHandler<PacketCompassRequest, AppEngPacket>, ICompassCallback
 {
 
-	public final long attunement;
-	public final int cx;
-	public final int cz;
-	public final int cdy;
-
-	EntityPlayer talkBackTo;
+	private long attunement;
+	private int cx;
+	private int cz;
+	private int cdy;
+	private EntityPlayer talkBackTo;
 
 	// automatic.
-	public PacketCompassRequest( ByteBuf stream )
+	public PacketCompassRequest()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)